pkg/users/password-dialogs.js

```diff
@@ -0,0 +1,327 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from 'react';
+import { superuser } from "superuser.jsx";
+
+import { Validated } from "./dialog-utils.js";
+import { show_modal_dialog } from "cockpit-components-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+function passwd_self(old_pass, new_pass) {
+    var old_exps = [
+        /Current password: $/,
+        /.*\(current\) UNIX password: $/,
+    ];
+    var new_exps = [
+        /.*New password: $/,
+        /.*Retype new password: $/,
+        /.*Enter new \w*\s?password: $/,
+        /.*Retype new \w*\s?password: $/
+    ];
+    var bad_exps = [
+        /.*BAD PASSWORD:.*/
+    ];
+    var too_new_exps = [
+        /.*must wait longer to change.*/
+    ];
+
+    var dfd = cockpit.defer();
```

Comment:
While we are at code modernization, this could also move to a standard `Promise`.

Reply:
Done.